solvers.go

```diff
@@ -898,6 +919,232 @@ func (s *VanillaSolver) Step(model []ValueGrad) (err error) {
 	return
 }
 
+// MomentumSGDSolver is the stochastic gradient descent optimizer with momentum item.
+type MomentumSGDSolver struct {
```

Comment:
`type Momentum struct`

Because in v0.10.0 all the solvers will be moved to their own package.. anything new should follow the new scheme